Src/BootCamp.Chapter/Demo/DemoApplication.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using BootCamp.Chapter.ApplicationState;
+using BootCamp.Chapter.UserInput;
+
+namespace BootCamp.Chapter.Demo
+{
+    public class DemoApplication
+    {
+        public IApplicationStateController ApplicationStateController { get; }
```

Comment:
Why don't you just put `ApplicationStarted` event here?

Reply:
I'd remove the state controller.

Reply:
I've removed the ApplicationStateController class as a whole. It makes more sense for the DemoApplication as an object to manage/monitor its internal state.

For the second point though, I have decided to avoid adding events to the ctor. I wanted to work with a single ApplicationStateChanged event instead of using multiple. If I'd want to fire the event inside the ctor, I'd have to make it static and initialize it before an objects are instantiated from the outside. Otherwise, if I'd keep them as non-static members, I'd have to subscribe to them inside the ctor, otherwise, they'd be just null open initialization.

Maybe I got that one wrong there but that's what worked out for me. :)